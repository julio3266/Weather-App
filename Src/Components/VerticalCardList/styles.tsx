import styled from "styled-components/native";

export const Container = styled.View`
  top: 17%;
  width: 100%;
  height: 40%;
  max-height: 40%;
  background-color: ${({ theme }) => theme.colors.blackTransparence};
  border-radius: 20px;
`;

export const List = styled.FlatList`
  flex: 1;
`;

export const Row = styled.View`
  flex-direction: row;
  justify-content: space-around;
  align-items: center;
`;

export const ListTitle = styled.Text`
  color: ${({ theme }) => theme.colors.white};
  font-family: ${({ theme }) => theme.fonts.regular};
  font-size: 20px;
  padding: 6%;
`;

export const ListSubtitle = styled.Text`
  color: #fff;
  font-family: ${({ theme }) => theme.fonts.regular};
  font-size: 20px;
  width: 39.3%;
  padding: 2%;
  font-weight: 500;
`;

export const ListSubtitleSecondary = styled.Text`
  color: ${({ theme }) => theme.colors.white};
  font-size: 20px;
  font-weight: 500;
`;

export const Icon = styled.Image`
  width: 50px;
  height: 50px;
`;
