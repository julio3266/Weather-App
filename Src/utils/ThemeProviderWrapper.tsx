import React from "react";
import styled, { ThemeProvider } from "styled-components/native";
import theme from "../Global/theme";

interface IProps {
  children: any;
  padding?: boolean;
}

export const ThemeProviderWrapper: React.FC<IProps> = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};
