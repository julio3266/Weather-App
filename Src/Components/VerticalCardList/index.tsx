import React from "react";
import { assignTestId } from "../../utils/qualityAssurance";
import * as Styled from "./styles";

export interface IVerticalListProps {
  dailyName: string;
  tempMin: number;
  tempMax: number;
  situationPerDaily: string;
  icon: string;
}
export interface IVerticalCardList {
  testID?: string;
  IVerticalListData: IVerticalListProps[];
}

export const VerticalCardList: React.FC<IVerticalCardList> = ({
  testID = "VerticalCardList",
  IVerticalListData,
}) => {
  return (
    <Styled.Container {...assignTestId("View", testID)}>
      <Styled.List
        {...assignTestId("View", `${testID}_list`)}
        data={IVerticalListData}
        ListHeaderComponent={() => (
          <Styled.ListTitle {...assignTestId("Text", `${testID}_title`)}>
            Previs√£o para 7 dias
          </Styled.ListTitle>
        )}
        renderItem={({ item, index }) => (
          <Styled.Row>
            <Styled.ListSubtitle
              {...assignTestId("Text", `${testID}_subTitle_${index}`)}
            >
              {item.dailyName}
            </Styled.ListSubtitle>
            <Styled.ListSubtitleSecondary
              {...assignTestId("Text", `${testID}_subTitleSecondary_${index}`)}
            >
              <Styled.Icon
                {...assignTestId("Image", `${testID}_icon_${index}`)}
                source={{
                  uri: `http://openweathermap.org/img/wn/${item.icon}@4x.png`,
                }}
              />
            </Styled.ListSubtitleSecondary>
            <Styled.ListSubtitleSecondary
              {...assignTestId("Text", `${testID}_subTitleSecondary_${index}`)}
            >
              {item.tempMin}
            </Styled.ListSubtitleSecondary>
            <Styled.ListSubtitleSecondary
              {...assignTestId("Text", `${testID}_subTitleSecondary_${index}`)}
            >
              {item.tempMax}
            </Styled.ListSubtitleSecondary>
          </Styled.Row>
        )}
      />
    </Styled.Container>
  );
};
