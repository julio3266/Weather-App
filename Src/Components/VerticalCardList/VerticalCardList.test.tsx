import React from "react";
import { render } from "@testing-library/react-native";
import { mockTestID } from "../../utils/qualityAssurance";
import { ThemeProviderWrapper as wrapper } from "../../utils/ThemeProviderWrapper";
import { VerticalCardList, IVerticalCardList } from "../VerticalCardList";

describe("Components/Background", () => {
  let props: IVerticalCardList = {
    IVerticalListData: [
      {
        dailyName: "segunda-feira",
        icon: "20d",
        situationPerDaily: "nublado",
        tempMax: 35,
        tempMin: 30,
      },
      {
        dailyName: "segunda-feira",
        icon: "20d",
        situationPerDaily: "nublado",
        tempMax: 35,
        tempMin: 30,
      },
    ],
  };
  const { findByTestId, toJSON } = render(<VerticalCardList {...props} />, {
    wrapper,
  });

  const sut_container = mockTestID("View", props.testID!).testID;
  const el_list = mockTestID("View", `${props.testID!}_list`).testID;
  const el_title = mockTestID("Text", `${props.testID!}_title`).testID;
  const el_subTitle = mockTestID("Text", `${props.testID!}_subTitle`).testID;
  const el_subTitleSecondary = mockTestID(
    "Text",
    `${props.testID!}_subTitleSecondary`
  ).testID;

  const el_icon = mockTestID("Image", `${props.testID!}_icon`).testID;

  it(`should render #${sut_container}`, () => {
    const sut = findByTestId(sut_container);
    expect(sut).toBeTruthy();
  });

  it(`should render #${el_list}`, () => {
    const sut = findByTestId(el_list);
    expect(sut).toBeTruthy();
  });

  it(`should render #${el_title}`, () => {
    const sut = findByTestId(el_title);
    expect(sut).toBeTruthy();
  });

  it(`should render #${el_subTitle}`, () => {
    const sut = findByTestId(el_subTitle);
    expect(sut).toBeTruthy();
  });

  it(`should render #${el_subTitle}`, () => {
    const sut = findByTestId(el_subTitle);
    expect(sut).toBeTruthy();
  });

  it(`should render #${el_subTitleSecondary}`, () => {
    const sut = findByTestId(el_subTitleSecondary);
    expect(sut).toBeTruthy();
  });

  it(`should render #${el_icon}`, () => {
    const sut = findByTestId(el_icon);
    expect(sut).toBeTruthy();
  });

  it(`should render a snapshot`, () => {
    expect(toJSON()).toMatchSnapshot();
  });
});
