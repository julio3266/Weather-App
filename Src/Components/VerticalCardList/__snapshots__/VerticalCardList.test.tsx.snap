// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Background should render a snapshot 1`] = `null`;
